library/src/main/java/com/wdullaer/materialdatetimepicker/date/DatePickerDialog.java

```diff
@@ -295,6 +299,11 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
             mCancelString = savedInstanceState.getString(KEY_CANCEL_STRING);
         }
 
+        if (mShowYearOnly) {
```

Comment:
Rather than removing these views, set their visibility to GONE. 
This avoids potential `NullPointerExceptions` when other pieces of the code try to fetch them